das/storage_service.go

```diff
@@ -23,6 +23,7 @@ type StorageService interface {
 	Closer
 	fmt.Stringer
 	HealthCheck(ctx context.Context) error
+	putKeyValue(ctx context.Context, key common.Hash, value []byte) error
```

Comment:
The IPFS storage service I'm working on wont be able to implement `putKeyValue` since the key in IPFS must be determined from the value. 

A way that you could do it is that you could make a separate interface with `putKeyValue` that wraps `StorageService`. `IterableStorageService` could use that interface instead, so that only objects that support `putKeyValue` can be used with it. For the other uses, eg `RedundantStorageService` and the cache related uses, you could try to cast to the new interface when `putKeyValue` is called. There may be other ways too.

Reply:
Done.